anchor/common/qbft/src/lib.rs

```diff
@@ -905,22 +920,52 @@ where
         data_hash: D::Hash,
         round_change_justification: Vec<SignedSSVMessage>,
         prepare_justification: Vec<SignedSSVMessage>,
+        round: Option<Round>,
     ) -> UnsignedWrappedQbftMessage {
         let data = self.get_message_data(&msg_type, data_hash);
 
+        let round = if let Some(round) = round {
+            round
+        } else {
+            data.round.into()
+        };
+
+        // Clear full_data from justifications as these do not store full data.
```

Comment:
I added this. 

Justifications are SignedMessages, but they should not store FullData. I wasn't super familiar with the code base. It kinda feels like we should handle this at the type-level and have separate types to avoid any possible chance of mixing this up. 

Anyway, for the spec tests, we have to strip the full data bytes off them when creating an unsigned message.